### ISSUE TITLE
Improve formatting of "spack find"

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#460]
Apparently, if I have boost anywhere on my system, everything else gets triple-spaced: the boost line is super-long, and all the other lines are space-padded at the end to match it.

I think this would be better if it were just printed out one package per line, without any attempts to format or pad.

**Comment by adamjstewart** [comment]
I agree that the triple-spaced thing is annoying. But I did like it a lot when it listed multiple packages on one line (back before I installed boost and mvapich2). Here are some alternative suggestions:

1. package
2. package@version
3. package@version+only+activated+variants
4. package@version+all~variants+even~unactivated (what we currently do)

I guess ideally, it would be nice if it listed all installed packages, even if multiple versions and variants of the same package are installed. Maybe we could print the bare minimum to make these distinguishable? Like if all installed variants of boost have the same 20 variants activated, but one has a different one, just list that one. Or if all versions of mpich are the same, don't show the version. Or if only one copy of hdf is installed, just print hdf. That would make uninstallation easier I think.
